pypeit/spectrographs/ldt_deveny.py

```diff
@@ -0,0 +1,399 @@
+            Binning in lois headers is space-separated rather than comma-separated.
+            """
+            binspec, binspatial = headarr[0]['CCDSUM'].split()
+            binning = parse.binning2string(binspec, binspatial)
+            return binning
+        
+        elif meta_key == 'mjd':
+            """
+            Use astropy to convert 'DATE-OBS' into a mjd.
+            """
+            ttime = Time(headarr[0]['DATE-OBS'], format='isot')
+            return ttime.mjd
+        
+        elif meta_key == 'lampstat01':
+            """
+            The spectral comparison lamps turned on are listed in `LAMPCAL`, but
+            if no lamps are on, then this string is blank.  Return either the
+            populated `LAMPCAL` string, or 'off' to ensure a positive entry for
+            `lampstat01`.
+            """
+            lampcal = headarr[0]['LAMPCAL'].strip()
+            if lampcal == '':
+                return 'off'
+            else:
+                return lampcal
+        
+        elif meta_key == 'dispname':
+            """
+            Convert older FITS keyword GRATING (gpmm/blaze) into the newer
+            Grating ID names (DVx) for easier identification of disperser.
+            """
+            gratings = {"DV1": "150/5000", "DV2": "300/4000", "DV3": "300/6750",
+                        "DV4": "400/8500", "DV5": "500/5500", "DV6": "600/4900",
+                        "DV7": "600/6750", "DV8": "831/8000", "DV9": "1200/5000",
+                        "DV10": "2160/5000", "DVxx": "UNKNOWN"}
+            ids = list(gratings.keys())
+            kwds = list(gratings.values())
+            idx = kwds.index(headarr[0]['GRATING'])
+            return f"{ids[idx]} ({kwds[idx]})"
+        
+        elif meta_key == 'decker':
+            """
+            Provide a stub for future inclusion of a decker on LDT/DeVeny.
+            """
+            if "DECKER" in headarr[0].keys():
+                decker = headarr[0]['DECKER']
+            else:
+                decker = "None"
+            return decker
+
+        elif meta_key == 'filter1':
+            """
+            Remove the parenthetical knob position to leave just the filter name
+            """
+            return headarr[0]['FILTREAR'].split()[0].upper()
+
+        else:
+            msgs.error("Not ready for this compound meta for LDT/DeVeny")
+
+    @classmethod
+    def default_pypeit_par(cls):
+        """
+        Return the default parameters to use for this instrument.
+        
+        Returns:
+            :class:`~pypeit.par.pypeitpar.PypeItPar`: Parameters required by
+            all of ``PypeIt`` methods.
+        """
+        par = super().default_pypeit_par()
+
+        # Calibration Parameters
+        # Turn off illumflat -- other defaults OK (as of v1.4.1)
+        set_use = dict(use_illumflat=False)
+        par.reset_all_processimages_par(**set_use)
+
+        # Use median combine (rather than weighted mean) for all cals
```

Comment:
maybe explain why

Reply:
Hrm, "because I've always just done it that way" is not a very good justification.  Upon careful consideration, changed all calibration combines except `pixelflatframe` to `weightmean`.  Added brief explanation to comment in code.